### DISCUSSION TITLE
does pulumi yaml have a `for_each` / `count` (terraform) equivalent?

### DISCUSSION THREAD

**Discussion by foxydevloper** [pulumi/pulumi#12560] [Pulumi in Practice] [answered]
Moving my infrastructure to use pulumi instead of terraform, not a fan of any of the provided languages, so I'm using pulumi yaml. Is there any equivalent to terraform's `for_each` / `count` in the pulumi yaml? I'm trying to deploy a server in multiple regions. The second best thing I've found is to make a stack for each region, although I'd want to be able to deploy to every region with a single command

**Comment by AaronFriel** [comment, marked as answer]
@foxydevloper thanks for asking. No, Pulumi YAML does not have any for loop or conditional logic. You have a few options, including your workaround.

You could use a tool to generate YAML, such as cue, jsonnet, or even a shell script. The `compiler` option allows you to specify an external command to generate the program:

https://www.pulumi.com/blog/extending-pulumi-languages-with-yaml-cue-jsonnet-rust/

https://leebriggs.co.uk/blog/2022/05/04/deploying-kubernetes-clusters-in-absurd-languages

You could also convert the YAML program to another language and use the looping functionality in it, via `pulumi convert`:

https://www.pulumi.com/docs/reference/cli/pulumi_convert/

**Reply by foxydevloper** [reply]
Thank you! I've used jsonnet before, so I'll go with that.

**Comment by ericrudder** [comment]
Glad you have a path forward.
